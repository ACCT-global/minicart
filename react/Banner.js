import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'render'

class Banner extends Component {
  static propTypes = {
    altText: PropTypes.string,
    image: PropTypes.string,
    page: PropTypes.string,
    targetParams: PropTypes.object,
  }

  static schema = {
    component: 'Banner',
    description: 'A simple banner component',
    type: 'object',
    properties: {
      alt: {
        type: 'string',
        title: 'Alt text',
      },
      image: {
        type: 'string',
        title: 'Image',
      },
      page: {
        type: 'string',
        title: 'Link',
      },
      targetParams: {
        type: 'object',
        title: 'Target params',
        properties: {
          params: {
            type: 'string',
            title: 'Params',
          },
        },
      },
    },
  }

  render() {
    const { altText, image, page, targetParams } = this.props
    return (
      <div className="flex items-center">
        <Link className="mb0" page={page} params={targetParams}>
          <img src={image} alt={altText} />
        </Link>
      </div>
    )
  }
}

export default Banner
