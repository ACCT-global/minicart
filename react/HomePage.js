import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { ExtensionPoint } from 'render'

export default class HomePage extends Component {
  static contextTypes = {
    prefetchPage: PropTypes.func,
  }

  componentDidMount() {
    this.context.prefetchPage('store/product')
  }

  render() {
    return (
      <div className="w-100 h-100">
        <ExtensionPoint id="carousel" />
        <ExtensionPoint id="shelf-top" />
        <ExtensionPoint id="banner" />
        <ExtensionPoint id="shelf-mid" />
      </div>
    )
  }
}
