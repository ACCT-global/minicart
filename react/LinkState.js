const gql = require('graphql-tag')

const minicartItemsQuery = gql`
  query {
    minicart @client {
      items {
        id
        name
        imageUrl
        detailUrl
        skuName
        quantity
        sellingPrice
        listPrice
        seller
        index
      }
    }
  }
`

const resolvers = {
  Mutation: {
    addToCart: (_, { items }, { cache }) => {
      const query = minicartItemsQuery
      const {
        minicart: { items: prevItems },
      } = cache.readQuery({ query })

      const indexedItems = items.map(item => ({
        index: prevItems.findIndex(({ id }) => id === item.id),
        item,
      }))

      const newItems = []
      for (const indexedItem of indexedItems) {
        const { index, item } = indexedItem
        if (index !== -1) {
          prevItems[index] = item
        } else {
          newItems.push(item)
        }
      }

      cache.writeData({
        data: {
          minicart: {
            __typename: 'Minicart',
            items: newItems.concat(prevItems).map(mapToMinicartItem),
          },
        },
      })
      return newItems
    },
    updateItems: (_, { items: newItems }, { cache }) => {
      const query = minicartItemsQuery
      const {
        minicart: { items: prevItems },
      } = cache.readQuery({ query })

      const items = prevItems
        .map(prevItem => {
          const newItem = newItems.find(({ id }) => id === prevItem.id)
          return newItem || prevItem
        })
        .map(mapToMinicartItem)

      cache.writeData({
        data: {
          minicart: { __typename: 'Minicart', items },
        },
      })
      return items
    },
    fillCart: (_, { items: newItems }, { cache }) => {
      const items = newItems.map(mapToMinicartItem)
      cache.writeData({
        data: {
          minicart: { __typename: 'Minicart', items },
        },
      })
      return items
    },
  },
}

const mapToMinicartItem = item => ({
  __typename: 'MinicartItem',
  seller: null,
  index: null,
  ...item,
})

const initialState = {
  minicart: { __typename: 'Minicart', items: [] },
}

module.exports = { resolvers, initialState }
