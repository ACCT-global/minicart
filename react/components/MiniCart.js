import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql, compose } from 'react-apollo'
import Button from '@vtex/styleguide/lib/Button'
import CloseIcon from '@vtex/styleguide/lib/icon/Close'
import orderFormQuery from '../queries/orderFormQuery.gql'
import updateItemsMutation from '../mutations/updateItemsMutation.gql'
import MiniCartItem from './MiniCartItem'

class MiniCart extends Component {
  static propTypes = {
    data: PropTypes.object,
    mutate: PropTypes.func,
  }

  handleRemoveItem = ({ target: { id } }) => {
    const { mutate, data: { orderForm } } = this.props
    const itemPayload = orderForm.items.find(item => item.id === id)
    const index = orderForm.items.indexOf(itemPayload)
    const updatedItem = [itemPayload].map(item => {
      return {
        id: parseInt(item.id),
        index: index,
        quantity: 0,
        seller: 1,
      }
    })

    mutate({
      variables: {
        orderFormId: orderForm.orderFormId,
        items: updatedItem,
      },
      refetchQueries: [{ query: orderFormQuery }],
    })
  }

  getItemId = detailUrl => {
    const regExp = /\/([^)]+)\//
    return regExp.exec(detailUrl)[1]
  }

  render() {
    const { data } = this.props
    if (data.loading) {
      return null
    }
    return (
      <div className="pa3">
        {data.orderForm.items.map(item => (
          <div className="flex flex-row" key={item.id}>
            <MiniCartItem
              imageUrl={item.imageUrl}
              name={item.name}
              price={item.sellingPrice}
              productId={this.getItemId(item.detailUrl)}
            />
            <Button id={item.id} onClick={this.handleRemoveItem}>
              <CloseIcon />
            </Button>
          </div>
        ))}
      </div>
    )
  }
}

export default compose(graphql(orderFormQuery), graphql(updateItemsMutation))(
  MiniCart
)
