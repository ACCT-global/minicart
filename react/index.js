import React from 'react'
import MiniCartButton from './MiniCartButton'

export default function HelloWorld() {
  return (
    <div className="relative w-100 bg-red h6">
      <div className="absolute right-0 mr6"><MiniCartButton labelButtonFinishShopping={'Finalizar Compra'} labelMiniCartEmpty={'Sua sacola estÃ¡ vazia!'} /></div>
    </div>
  )
}
