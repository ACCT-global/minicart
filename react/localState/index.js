import { head, mergeDeepRight, values } from 'ramda'

import {
  updateOrderFormShipping,
  updateOrderFormCheckin,
} from 'vtex.store-resources/Mutations'
import { minicartItemsQuery, minicartOrderFormQuery } from './queries'

export default function(client) {
  const replayOrderFormServerMutation = mutation => async (
    _,
    variables,
    { cache }
  ) => {
    const response = await client.mutate({
      variables,
      mutation,
    })

    const updatedOrderForm = head(values(response.data))

    const {
      minicart: { orderForm: outdatedOrderForm },
    } = cache.readQuery({ query: minicartOrderFormQuery })

    const orderForm = JSON.stringify(
      mergeDeepRight(JSON.parse(outdatedOrderForm), updatedOrderForm)
    )
    cache.writeData({
      data: {
        minicart: { __typename: 'Minicart', orderForm },
      },
    })
    return orderForm
  }

  const resolvers = {
    Mutation: {
      addToCart: (_, { items }, { cache }) => {
        const query = minicartItemsQuery
        const {
          minicart: { items: prevItems },
        } = cache.readQuery({ query })

        const newItems = items.map(item =>
          mapToMinicartItem({ ...item, upToDate: false })
        )
        const writeItems = [...JSON.parse(prevItems), ...newItems]

        console.log('teste add to cart writing: ', JSON.stringify(writeItems))
        try {
          cache.writeData({
            data: {
              minicart: {
                __typename: 'Minicart',
                items: JSON.stringify(writeItems),
              },
            },
          })
        } catch (err) {
          console.log('teste ERRO: ', err)
        }
        return writeItems
      },
      updateItems: (_, { items: newItems }, { cache }) => {
        // UPDATE ITEMS MUST PASS INDEX on newItems :)
        console.log('teste updateItems!!!')
        const query = minicartItemsQuery
        const {
          minicart: { items: prevItems },
        } = cache.readQuery({ query })

        const cleanNewItems = newItems.filter(({ index }) => index != null)
        const items = [...JSON.parse(prevItems)]

        for (let i = 0; i < cleanNewItems.length; i++) {
          const newItem = cleanNewItems[i]
          const { index } = newItem
          const prevItem = JSON.parse(prevItems)[index]
          items[index] = mapToMinicartItem(
            mergeDeepRight(prevItem, { ...newItem, upToDate: false })
          )
        }

        // const items = prevItems.map((prevItem, index) => {
        //   const
        //   // console.log('teste found newItem: ', newItem)
        //   const item = newItem
        //     ? mergeDeepRight(prevItem, { ...newItem, upToDate: false })
        //     : prevItem
        //   return mapToMinicartItem(item)
        // })
        // const items = prevItems.map((prevItem, index) => {
        //   const newItem = newItems.find(({ id }) => id === prevItem.id)
        // const item = newItem
        //   ? mergeDeepRight(prevItem, { ...newItem, upToDate: false })
        //   : prevItem
        // return mapToMinicartItem({ ...item, index })
        // })

        console.log('teste updateItems writing: ', items)
        cache.writeData({
          data: {
            minicart: { __typename: 'Minicart', items: JSON.stringify(items) },
          },
        })
        return JSON.stringify(items)
      },
      updateOrderForm: (_, { orderForm: newOrderForm }, { cache }) => {
        const orderForm = mapToLinkStateOrderForm(newOrderForm)
        console.log('teste orderForm: ', orderForm)
        cache.writeData({
          data: {
            minicart: {
              __typename: 'Minicart',
              orderForm: JSON.stringify(orderForm),
            },
          },
        })
        if (orderForm.items) {
          updateLinkStateItems(orderForm.items, cache)
        }
        return JSON.stringify(orderForm)
      },
      updateOrderFormShipping: replayOrderFormServerMutation(
        updateOrderFormShipping
      ),
      updateOrderFormCheckin: replayOrderFormServerMutation(
        updateOrderFormCheckin
      ),
    },
  }

  const updateLinkStateItems = (newItems, cache) => {
    const items = newItems.map(item =>
      mapToMinicartItem({
        ...item,
        upToDate: true,
      })
    )
    console.log('teste updateLinkStateItems writing: ', items)

    cache.writeData({
      data: {
        minicart: { __typename: 'Minicart', items: JSON.stringify(items) },
      },
    })
    return items
  }

  const mapToLinkStateOrderForm = orderForm => ({
    ...orderForm,
    __typename: 'OrderFormClient',
    items: orderForm.items && orderForm.items.map(mapToOrderFormItem),
    totalizers:
      orderForm.totalizers && orderForm.totalizers.map(mapToOrderFormTotalizer),
    clientProfileData: orderForm.clientProfileData && {
      ...orderForm.clientProfileData,
      __typename: 'OrderFormClientProfileData',
    },
    shippingData:
      orderForm.shippingData &&
      mapToOrderFormShippingData(orderForm.shippingData),
    storePreferencesData:
      orderForm.storePreferencesData &&
      mapToOrderFormStorePreferences(orderForm.storePreferencesData),
    // itemMetadata: orderForm.itemMetadata ? mapToOrderFormStoreItemMetadata(orderForm.itemMetadata) : null,
  })

  // id
  //           name
  //           skuName
  //           productId
  //           refId
  //           ean
  //           imageUrl
  //           detailUrl

  const mapToItemMetdataItem = itemMetdataItem => ({
    detailUrl: null,
    ean: null,
    id: null,
    imageUrl: null,
    skuName: null,
    name: null,
    productId: null,
    refId: null,
    ...itemMetdataItem,
    assemblyOptions: (itemMetdataItem.assemblyOptions || []).map(
      mapToMetadataAssemblyOptionItem
    ),
    __typename: 'OrderFormClientMetadataItem',
  })

  const mapToMetadataAssemblyOptionItem = MetadataAssemblyOption => ({
    id: null,
    name: null,
    required: null,
  })

  const mapToOrderFormStoreItemMetadata = itemMetadata => ({
    ...itemMetadata,
    __typename: 'OrderFormClientItemMetadata',
    items: itemMetadata.items.map(mapToItemMetdataItem),
  })

  const mapAssemblyOptions = (options = {}) => ({
    parentPrice: null,
    ...options,
    added: (options.added || []).map(added => ({
      ...added,
      item: added.item && mapToOrderFormItem(added.item),
      __typename: 'AddedAssemblyOptions',
    })),
    removed: (options.removed || []).map(option => ({
      ...option,
      __typename: 'RemovedAssemblyOption',
    })),
    __typename: 'AssemblyOptions',
  })

  const mapToOrderFormItem = item => ({
    ...item,
    assemblyOptions: mapAssemblyOptions(item.assemblyOptions),
    __typename: 'OrderFormItem',
  })

  const mapToOrderFormTotalizer = totalizer => ({
    ...totalizer,
    __typename: 'OrderFormTotalizer',
  })

  const mapToOrderFormShippingData = ({ address, availableAddresses }) => ({
    address: address && mapToAddress(address),
    availableAddresses:
      availableAddresses && availableAddresses.map(mapToAddress),
    __typename: 'OrderFormShippingData',
  })

  const mapToAddress = address => ({
    ...address,
    __typename: 'Address',
  })

  const mapToOrderFormStorePreferences = storePreferencesData => ({
    ...storePreferencesData,
    __typename: 'OrderFormStorePreferencesData',
  })

  const mapItemOptions = option => ({
    seller: null,
    assemblyId: null,
    id: null,
    quantity: null,
    ...option,
    __typename: 'MinicartItemOptions',
  })

  const mapToMinicartItem = item => ({
    seller: null,
    index: null,
    parentItemIndex: null,
    parentAssemblyBinding: null,
    imageUrl: null,
    id: null,
    name: null,
    detailUrl: null,
    skuName: null,
    quantity: null,
    sellingPrice: null,
    listPrice: null,
    cartIndex: null,
    ...item,
    options: item.options ? item.options.map(mapItemOptions) : [],
    assemblyOptions: mapAssemblyOptions(item.assemblyOptions),
    __typename: 'MinicartItem',
  })

  const initialState = {
    minicart: {
      __typename: 'Minicart',
      items: '[]',
      orderForm: null,
    },
  }

  return { resolvers, initialState }
}
