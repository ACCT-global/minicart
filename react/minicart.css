.contentSmall {
  max-height: 383px;
}

.badge {
  top: -0.7rem;
  right: -0.8rem;
}

.arrowUp {
  right: 7px;
}

.box {
  width: 364px;
  top: 100%;
}

.itemImageLoader {
  width: 100%;
  height: 100%;
}

.contentFooter {
  margin-top: auto;
}

@media only screen and (max-width: 40rem) {
  body.sidebarOpen {
    overflow: hidden;
  }
}

@media only screen and (min-width: 40em) {
  .sidebar {
    width: 340px;
  }
}
