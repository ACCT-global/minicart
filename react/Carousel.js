import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Slider from 'react-slick'
import { NoSSR } from 'render'

import Banner from './Banner'

class Carousel extends Component {
  static propTypes = {
    banner1: PropTypes.shape({
      image: PropTypes.string.isRequired,
      page: PropTypes.string,
    }),
    banner2: PropTypes.shape({
      image: PropTypes.string.isRequired,
      page: PropTypes.string,
    }),
  }

  static schema = {
    component: 'Carousel',
    description: 'A simple carousel component',
    type: 'object',
    properties: {
      banner1: {
        type: 'object',
        title: 'Banner 1',
        properties: {
          image: {
            type: 'string',
            title: 'Banner 1 image',
          },
          page: {
            type: 'string',
            title: 'Banner 1 link',
          },
          targetParams: {
            type: 'object',
            title: 'Banner 1 target params',
            properties: {
              params: {
                type: 'string',
                title: 'Params',
              },
            },
          },
        },
      },
      banner2: {
        type: 'object',
        title: 'Banner 2',
        properties: {
          image: {
            type: 'string',
            title: 'Banner 2 image',
          },
          page: {
            type: 'string',
            title: 'Banner 2 link',
          },
          targetParams: {
            type: 'object',
            title: 'Banner 2 target params',
            properties: {
              params: {
                type: 'string',
                title: 'Params',
              },
            },
          },
        },
      },
    },
  }

  render() {
    const { banner1, banner2 } = this.props
    const settings = {
      dots: false,
      infinite: true,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
    }

    const fallback = (
      <Banner
        image={banner1.image}
        altText={banner1.page}
        page={banner1.page}
        targetParams={banner1.targetParams}
      />
    )

    return (
      <NoSSR onSSR={fallback}>
        <Slider {...settings}>
          <div>
            <Banner
              image={banner1.image}
              altText={banner1.page}
              page={banner1.page}
              targetParams={banner1.targetParams}
            />
          </div>
          <div>
            <Banner
              image={banner2.image}
              altText={banner2.page}
              page={banner2.page}
              targetParams={banner2.targetParams}
            />
          </div>
        </Slider>
      </NoSSR>
    )
  }
}

export default Carousel
