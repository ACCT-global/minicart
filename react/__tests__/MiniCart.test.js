import React from 'react'
import { render, fireEvent } from '@vtex/test-tools/react'
import wait from 'waait'

import MiniCart from './../index'

describe('<MiniCart /> component', () => {
  async function resolveApolloQueries() {
    // Server queries
    await wait(0)
    // Local state queries
    await wait(0)
  }

  it('should be rendered', () => {
    const { asFragment } = render(<MiniCart type="popup" hideContent={false} />)
    expect(asFragment()).toBeDefined()
  })

  it('should render popup onClick', () => {
    const { getByText, baseElement } = render(
      <MiniCart type="popup" hideContent={false} />
    )

    let box = baseElement.querySelector('.box')
    let sidebar = baseElement.querySelector('.sidebar')

    // Expect box be null before click
    expect(box).toBeNull()
    // Sidebar should not exist before and after click
    expect(sidebar).toBeNull()

    fireEvent.click(getByText('Button Test'))

    box = baseElement.querySelector('.box')
    sidebar = baseElement.querySelector('.sidebar')

    // Expect Minicart Popup redered
    expect(box).toBeDefined()
    expect(box).not.toBeNull()
    // Expect sidebar still not rendered
    expect(sidebar).toBeNull()
  })

  it('should display sidebar onClick', () => {
    const { getByText, baseElement } = render(
      <MiniCart type="sidebar" hideContent={false} />
    )

    let box = baseElement.querySelector('.box')
    let sidebar = baseElement.querySelector('.sidebarScrim.dn')

    // Expect box be null before and after click
    expect(box).toBeNull()
    // Sidebar should have diplay none class before click
    expect(sidebar).toBeDefined()
    expect(sidebar).not.toBeNull()

    fireEvent.click(getByText('Button Test'))

    box = baseElement.querySelector('.box')
    sidebar = baseElement.querySelector('.sidebarScrim.dn')

    // Expect sidebar dont have display none after click
    expect(sidebar).toBeNull()
    expect(box).toBeNull()
  })

  it('should match the snapshot in popup mode', async () => {
    const leftClick = { button: 0 }

    const { getByText, asFragment } = render(
      <MiniCart type="popup" hideContent={false} />
    )

    await resolveApolloQueries()

    fireEvent.click(getByText('Button Test'), leftClick)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot in sidebar mode', async () => {
    const { baseElement } = render(
      <MiniCart type="sidebar" hideContent={false} />
    )

    await resolveApolloQueries()
    expect(baseElement).toMatchSnapshot()
  })
})
