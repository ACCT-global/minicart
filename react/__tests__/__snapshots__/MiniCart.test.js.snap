// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiniCart /> component should match the snapshot in popup mode 1`] = `
<DocumentFragment>
  <aside
    class="container relative fr flex items-center"
  >
    <button
      variation="tertiary"
    >
      Button Test
      <span
        class="flex items-center"
      >
        <span
          class="relative gray"
        >
          <svg
            class="undefined IconCart"
          >
            <rect />
          </svg>
          <span
            class="badge c-on-emphasis absolute t-mini bg-emphasis br4 w1 h1 pa1 flex justify-center items-center lh-solid"
          >
            3
          </span>
        </span>
      </span>
    </button>
    <div>
      <div
        class="box dn db-ns absolute z-max flex flex-colunm"
        style="right: 0px;"
      >
        <div
          class="shadow-3"
        >
          <div
            class="arrowUp absolute top-0 shadow-3 bg-base h1 w1 pa4 rotate-45"
          />
          <div
            class="mt3 bg-base relative flex flex-column"
          >
            <div
              class="content overflow-x-hidden pa1 contentSmall bg-base overflow-y-scroll"
            >
              <section
                class="relative flex"
              >
                <div
                  class="fr absolute top-0 right-0"
                >
                  <button
                    variation="tertiary"
                  >
                    Button Test
                    <svg
                      class="undefined IconDelete"
                      height="15"
                      width="15"
                    >
                      <rect
                        height="15"
                        width="15"
                      />
                    </svg>
                  </button>
                </div>
                <div>
                   Extension Point 
                </div>
              </section>
              <section
                class="relative flex"
              >
                <div
                  class="fr absolute top-0 right-0"
                >
                  <button
                    variation="tertiary"
                  >
                    Button Test
                    <svg
                      class="undefined IconDelete"
                      height="15"
                      width="15"
                    >
                      <rect
                        height="15"
                        width="15"
                      />
                    </svg>
                  </button>
                </div>
                <div>
                   Extension Point 
                </div>
              </section>
              <section
                class="relative flex"
              >
                <div
                  class="fr absolute top-0 right-0"
                >
                  <button
                    variation="tertiary"
                  >
                    Button Test
                    <svg
                      class="undefined IconDelete"
                      height="15"
                      width="15"
                    >
                      <rect
                        height="15"
                        width="15"
                      />
                    </svg>
                  </button>
                </div>
                <div>
                   Extension Point 
                </div>
              </section>
            </div>
            <div
              class="contentFooter w-100 bg-base pa4 pv5 flex flex-column items-end"
            >
              <div
                class="contentPrice mb3"
              >
                <div
                  class="price-mock"
                >
                  Price
                </div>
              </div>
              <div
                class=""
              >
                <button
                  variation="primary"
                >
                  Button TestGo to checkout
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;

exports[`<MiniCart /> component should match the snapshot in sidebar mode 1`] = `
<body>
  <div>
    <aside
      class="container relative fr flex items-center"
    >
      <button
        variation="tertiary"
      >
        Button Test
        <span
          class="flex items-center"
        >
          <span
            class="relative gray"
          >
            <svg
              class="undefined IconCart"
            >
              <rect />
            </svg>
            <span
              class="badge c-on-emphasis absolute t-mini bg-emphasis br4 w1 h1 pa1 flex justify-center items-center lh-solid"
            >
              3
            </span>
          </span>
        </span>
      </button>
    </aside>
  </div>
  <div>
    <div
      class="sidebarScrim fixed dim bg-base--inverted top-0 left-0 z-9999 w-100 vh-100 o-40 dn"
    />
    <div
      class="animation-mock"
    >
      <div
        class="sidebarHeader pointer flex flex-row items-center pa5 h3 bg-base w-100 z-max bb b--muted-3 bw1"
      >
        <div
          class="c-muted-1 pa4 flex items-center"
        >
          <svg
            class="undefined IconCaret"
            height="17"
            width="17"
          >
            <rect
              height="17"
              width="17"
            />
          </svg>
        </div>
        <aside
          class="container relative fr flex items-center"
        >
          <button
            variation="tertiary"
          >
            Button Test
            <span
              class="flex items-center"
            >
              <span
                class="relative c-muted-1"
              >
                <svg
                  class="undefined IconCart"
                >
                  <rect />
                </svg>
                <span
                  class="badge c-on-emphasis absolute t-mini bg-emphasis br4 w1 h1 pa1 flex justify-center items-center lh-solid"
                >
                  3
                </span>
              </span>
              <span
                class="label dn-m db-l t-action--small c-muted-1 pl6"
              >
                My mini cart
              </span>
            </span>
          </button>
        </aside>
      </div>
      <div
        class="content overflow-x-hidden pa1 contentLarge"
      >
        <section
          class="relative flex"
        >
          <div
            class="fr absolute top-0 right-0"
          >
            <button
              variation="tertiary"
            >
              Button Test
              <svg
                class="undefined IconDelete"
                height="15"
                width="15"
              >
                <rect
                  height="15"
                  width="15"
                />
              </svg>
            </button>
          </div>
          <div>
             Extension Point 
          </div>
        </section>
        <section
          class="relative flex"
        >
          <div
            class="fr absolute top-0 right-0"
          >
            <button
              variation="tertiary"
            >
              Button Test
              <svg
                class="undefined IconDelete"
                height="15"
                width="15"
              >
                <rect
                  height="15"
                  width="15"
                />
              </svg>
            </button>
          </div>
          <div>
             Extension Point 
          </div>
        </section>
        <section
          class="relative flex"
        >
          <div
            class="fr absolute top-0 right-0"
          >
            <button
              variation="tertiary"
            >
              Button Test
              <svg
                class="undefined IconDelete"
                height="15"
                width="15"
              >
                <rect
                  height="15"
                  width="15"
                />
              </svg>
            </button>
          </div>
          <div>
             Extension Point 
          </div>
        </section>
      </div>
      <div
        class="contentFooter w-100 bg-base pa4 pv5 flex flex-column items-end bt b--muted-3"
      >
        <div
          class="contentPrice mb3"
        >
          <div
            class="price-mock"
          >
            Price
          </div>
        </div>
        <div
          class="bb bw4 bw2-m b--transparent"
        >
          <button
            variation="primary"
          >
            Button Test
            Go to checkout
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
